driver-core/src/main/com/mongodb/connection/netty/NettyStream.java

```diff
@@ -336,4 +341,47 @@ public T get() throws IOException {
             }
         }
     }
+
+
+    private void scheduleReadTimeout() {
+        adjustTimeout(false);
+    }
+
+    private void disableReadTimeout() {
+        adjustTimeout(true);
+    }
+
+    private void adjustTimeout(final boolean disable) {
+            ChannelHandler timeoutHandler = channel.pipeline().get(READ_HANDLER_NAME);
+            if (timeoutHandler != null && timeoutHandler instanceof ReadTimeoutHandler) {
+                final ReadTimeoutHandler readTimeoutHandler = (ReadTimeoutHandler) timeoutHandler;
+                final ChannelHandlerContext handlerContext = channel.pipeline().context(timeoutHandler);
+                EventExecutor executor = handlerContext.executor();
+
+                if (disable) {
+                    if (executor.inEventLoop()) {
```

Comment:
Not directly -its essentially a race.  Any smoke test can using NettyStream can be observed using both branches.